spec/tlaw/namespace_spec.rb

```diff
@@ -91,6 +107,31 @@ module TLAW
           |  .some_ep(foo: nil)
         }.unindent)}
       end
+
+      context '.traverse' do
+        it 'can yield all children recursively' do
+          expect(namespace_class.traverse.to_a).to eq [
+            endpoint_class,
+            child_class,
+            child_endpoint_class,
+            grand_child_class
+          ]
+        end
+
+        it 'can yield all endpoints recursively' do
+          expect(namespace_class.traverse(:endpoints).to_a).to eq [
+            endpoint_class,
+            child_endpoint_class
+          ]
+        end
+
+        it 'can yield all namespaces recursively' do
+          expect(namespace_class.traverse(:namespaces).to_a).to eq [
+            child_class,
+            grand_child_class
+          ]
+        end
+      end
     end
```

Comment:
With [saharspec](https://github.com/zverok/saharspec) (already included) you can make it this way:
```ruby
subject { ->(*args) { namespace_class.traverse(*args).to_a } }

its_call { is_expected.to ret [...] }
its_call(:namespaces) { is_expected.to ret [...] }
its_call(:endpoints) { is_expected.to ret [...] }
# ...and also..
its_call(:garbage) { is_expected.to raise_error ArgumentError }
```
(`ret` is matcher name for "(its call) **returns**")

For me, this is more DRY and expresses "what's going on" better (params difference is not hidden in the middle of example, "what is test subject" also made obvious), but I will not insist, I heard different opinions on those tricks :)